pkg/artifactory/resource_artifactory_user.go

```diff
@@ -162,6 +162,11 @@ func resourceUserRead(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).ArtOld
 
 	user, resp, err := c.V1.Security.GetUser(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.